lambda/models/state_machine/create_model.py

```diff
@@ -14,27 +14,76 @@
 
 """Lambda handlers for CreateModel state machine."""
 
+import json
+import os
 from copy import deepcopy
 from typing import Any, Dict
 
+import boto3
+from models.domain_objects import CreateModelRequest
+
+lambdaClient = boto3.client("lambda")
+ecrClient = boto3.client("ecr")
+ec2Client = boto3.client("ec2")
+ddbResource = boto3.resource("dynamodb")
+model_table = ddbResource.Table(os.environ["MODEL_TABLE_NAME"])
+
 
 def handle_set_model_to_creating(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Set DDB entry to CREATING status."""
     output_dict = deepcopy(event)
-    output_dict["create_infra"] = True
+    request = CreateModelRequest.validate(event)
+
+    if (
+        request.AutoScalingConfig  # noqa: W503
+        and request.ContainerConfig  # noqa: W503
+        and request.InferenceContainer  # noqa: W503
+        and request.InstanceType  # noqa: W503
+        and request.LoadBalancerConfig  # noqa: W503
+    ):
+        model_table.update_item(
+            Key={"model_id": request.ModelId}, AttributeUpdates={"Status": {"Value": "CREATING", "Action": "PUT"}}
+        )
+        output_dict["create_infra"] = True
+    else:
+        output_dict["create_infra"] = False
     return output_dict
 
 
 def handle_start_copy_docker_image(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Start process for copying Docker image into local AWS account."""
     output_dict = deepcopy(event)
+    request = CreateModelRequest.validate(event)
+
+    response = lambdaClient.invoke(
+        FunctionName=os.environ["DOCKER_IMAGE_BUILDER_FN_ARN"],
+        Payload=json.dumps(
+            {
+                "base_image": request.ContainerConfig.BaseImage.BaseImage,
+                "layer_to_add": request.ContainerConfig.BaseImage.Path,
+            }
+        ),
+    )
+
+    payload = response["Payload"].read()
+    output_dict["image_info"] = json.loads(payload)
     return output_dict
 
 
 def handle_poll_docker_image_available(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Check that Docker image is available in account or not."""
     output_dict = deepcopy(event)
+
```

Comment:
Talked about over chat, but let's also add a field in the event/output_dict to hold a "poll_docker_iterations" field that can start at 0 and then get incremented in each iteration. once we exceed a certain amount of time (we wait a minute between polls), then we can just decide to stop iterating, clean up the ec2 instance, and raise an exception to fail the state machine

Reply:
added